### PR TITLE
dailyFees Ether-fi

### PR DESCRIPTION
**Pull request by 0xpeluche**
Add daily fees for `Ether-fi` following https://etherfi.gitbook.io/etherfi/liquid/technical-documentation#fees

### CODE REVIEW

**Comment by llamatester** [comment]
The ether-fi adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees ether-fi
    
    🦙 Running ETHER-FI adapter 🦙
    ---------------------------------------------------
    Start Date:	Fri, 05 Jul 2024 10:07:50 GMT
    End Date:	Sat, 06 Jul 2024 10:07:50 GMT
    ---------------------------------------------------
    
    ETHEREUM 👇
    Backfill start time: 12/3/2024
    Daily fees: 26.34 k
    End timestamp: 1720260469 (2024-07-06T10:07:49.000Z)

**Comment by llamatester** [comment]
The ether-fi adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees ether-fi
    
    🦙 Running ETHER-FI adapter 🦙
    ---------------------------------------------------
    Start Date:	Fri, 05 Jul 2024 10:56:23 GMT
    End Date:	Sat, 06 Jul 2024 10:56:23 GMT
    ---------------------------------------------------
    
    ETHEREUM 👇
    Backfill start time: 12/3/2024
    Daily fees: 27.10 k
    End timestamp: 1720263382 (2024-07-06T10:56:22.000Z)

File: fees/ether-fi/index.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  api.addTokens(asset, totalSupply);
+  return (await api.getUSDValue()) * dailyFeeRate;
+};
+
+const adapter: Adapter = {
+  version: 2,
+  adapter: {
+    [CHAIN.ETHEREUM]: {
+      fetch: async () => {
+        const api = new sdk.ChainApi({ chain: CHAIN.ETHEREUM });
```
Comment (dtmkeng) [inline fees/ether-fi/index.ts]: function `sdk.ChainApi` can't support backfill data, could you change use parma `options: FetchOptions`

and use ` options.api.call`
and use `  dailyFees.add(asset, totalSupply)` instend `api.addTokens`

**Review by dtmkeng — CHANGES_REQUESTED** [review]

File: fees/ether-fi/index.ts (hunk 1 above)
Comment (dtmkeng) [reply fees/ether-fi/index.ts]:
```
const fetch = async (options: FetchOptions) => {
  const dailyFeeRate = ANNUAL_FEE_RATE / YEAR;
  const dailyFees = options.createBalances();
  const [asset, totalSupply] = await Promise.all([
    options.api.call({
      target: LIQUID_VAULT,
      abi: "function asset() external view returns (address)",
    }),
    options.api.call({
      target: LIQUID_VAULT,
      abi: "function totalAssets() external view returns (uint256 assets)",
    }),
  ]);
  dailyFees.add(asset, totalSupply);
  dailyFees.resizeBy(dailyFeeRate);
  return { dailyFees }
}
```

**Comment by llamatester** [comment]
The ether-fi adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees ether-fi
    
    🦙 Running ETHER-FI adapter 🦙
    ---------------------------------------------------
    Start Date:	Fri, 05 Jul 2024 13:12:06 GMT
    End Date:	Sat, 06 Jul 2024 13:12:06 GMT
    ---------------------------------------------------
    
    ETHEREUM 👇
    Backfill start time: 12/3/2024
    Daily fees: 27.06 k
    End timestamp: 1720271525 (2024-07-06T13:12:05.000Z)

**Pull request merged**